ethcore/src/evm/interpreter/memory.rs

```diff
@@ -109,6 +114,20 @@ impl Memory for Vec<u8> {
 			Memory::resize(self, size)
 		}
 	}
+
+	fn into_return_data(mut self, offset: U256, size: U256) -> ReturnData {
+		let offset = offset.low_u64() as usize;
+		let size = size.low_u64() as usize;
+		if !is_valid_range(offset, size) {
```

Comment:
this just checks whether `offset + size` would overflow, but they should also be bounds-checked against `self.len()` at this point. that would also make the check within `ReturnData::new` redundant, so maybe it would be better as a precondition of that function.